vulpemventures/marina
# Lightning swaps

This PR implements submarine swaps between L-BTC and Lightning (using [Boltz](https://boltz.exchange/)):

- Receive L-BTC through Lightning (Reverse submarine swap)
- Send L-BTC to Lightning (Submarine swap)

@tiero please review

It closes https://github.com/vulpemventures/marina/issues/255

How to test:

[marina-0.4.9-RC3.zip](https://github.com/vulpemventures/marina/files/9002757/marina-0.4.9-RC3.zip)

Reverse submarine swap:

- On Marina click Receive
- Choose asset L-BTC
- Choose Lightning network
- Choose amount (between 0.0005 and ​​0.04294967 L-BTC)
- nsert password (needed to sign claim transaction)
- Copy invoice
- On [htlc.me](https://htlc.me/) choose Send
- Paste invoice on htlc.me
- Go back to Marina
- Invoice should be paid in less than 20 seconds

Submarine swap:

- On [htlc.me](https://htlc.me/) click Receive
- Enter amount (between 0.0005 and ​​0.04294967 L-BTC)
- Copy generated invoice
- On Marina click Send
- Choose asset L-BTC
- Choose Lightning network
- Paste invoice
- Proceed with payment
- Go back to [htlc.me](https://htlc.me/), payment should be paid in less than 20 seconds
